README.md

```diff
@@ -1,2 +1,202 @@
-# promise-utils
-Utilities for working with promises
+# `@consento/promise`
+
+A set of utilities for working with [Promises]() with particular focus
+on abort-ability using [`AbortSignal`s][AbortSignal].
+
+[Promise]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise
+[AbortSignal]: https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal
+
+## Goal
+
+Provide sturdy `Promise` utils that make it easier to write asynchronous code that is both performant
+and saves on unnecessary operations by making it easier to use AbortSignals.
+
+The library has a partial focus on `Typescript` utility as `TypeScript` comes with its own
+set of problems.
+
+## Contributions
+
+Additionally to the common Consento contribution guidelines, this repository is eager to be dissolved
+into standard projects. If you find one of the utils presented here useful in a more common context (e.g.
+Node.js repo, TypeScript or w3c) then we would find it awesome if you could promote it to find its way there so
+we can remove it from this library.
+
+We also appreciate pointers to projects that may offer a better solution than the solutions presented
+in this repo!
+
+## Utils
+
+- [`bubbleAbort()`](#bubble-abort) - tiny helper for dealing with abort signals
+- [`checkpoint()`](#checkpoint) - comfortable helper for dealing with abort signals
+- [`cbPromise()`](#cb-promise) - tiny helper for asynchronous code
+- [`cleanupPromise()`](#cleanup-promise) - versatily promise variant that allows react-style hook cleanups
+- [`composeAbort()`](#compose-abort) - compose an abort-signal into sub-variants.
+- [`isPromiseLike()`](#is-promise-like) - typescript type-check helper
+- [`raceWithSignal()`](#race-with-signal) - races several promises, all but the first will be aborted
+- [`toPromise()`](#to-promise) - turns a value into a promise, cleverly
+- [`wrapTimeout()`](#wrap-timeout) - wraps an async function to have an optional timeout
+
+#### `bubbleAbort()`
+
+Simple function that throws an AbortError if an AbortSignal
+happens to be aborted.
+
+_Usage:_
+
+```javascript
+const { bubbleAbort } = require('@consento/promise/bubbleAbort')
+
+async function longRunning ({ signal } = {}) {
+  let result = ''
+  for await (const data of internalIterator()) {
+    bubbleAbort(signal) // Will throw an error if the signal is aborted
+    result += data
+  }
+  return result
+}
+```
+
+#### `cbPromise()`
```

Comment:
This pattern is called a deferred promise. Might be good to use that name or use an existing deferred library?

Reply:
Cool, I didn't know (or probably forgot) how its called and searches didn't turn up results for me. Looking for a good deferred library.

Reply:
https://github.com/consento-org/promise/pull/1/commits/0331d079753da24d3108eadb16703e7972686081